autoblocks/_impl/global_state.py

```diff
@@ -25,10 +29,20 @@ def init() -> None:
         daemon=True,
     )
     background_thread.start()
-
+    signal.signal(signal.SIGTERM, _on_exit)
+    signal.signal(signal.SIGINT, _on_exit)
     _started = True
 
 
+def _on_exit(signum, frame):
+    if _loop and _loop.is_running():
+        log.info(f"Attempting to gracefully stop event loop. Received signal {signum}")
+        _loop.stop()
```

Comment:
is this right? we want to call stop before trying to clean up?

Reply:
Following this example: https://stackoverflow.com/questions/37417595/graceful-shutdown-of-asyncio-coroutines#:~:text=How%20can%20we%20wait%20tasks%20finished

Looking at this more closely though I think you are right

Reply:
hmm yeah from 2016, would be nice to find something more recent